services/kafka/service.go

```diff
@@ -159,16 +160,23 @@ func (c *Cluster) writer(target WriteTarget, diagnostic Diagnostic) (*writer, er
 		defer c.mu.Unlock()
 		w, ok = c.writers[topic]
 		if !ok {
-			wc, err := c.cfg.WriterConfig(diagnostic, target)
+			wc, err := c.cfg.writerConfig(diagnostic, target)
 			if err != nil {
 				return nil, err
 			}
-			kw := kafka.NewWriter(wc)
+			kp, err := kafka.NewAsyncProducer(c.cfg.Brokers, wc)
+
+			if err != nil {
+				return nil, err
+			}
+
 			// Create new writer
 			w = &writer{
-				kafka:   kw,
-				cluster: c.cfg.ID,
-				topic:   topic,
+				requestsInFlightMetric: metrics.GetOrRegisterCounter("requests-in-flight", wc.MetricRegistry),
+				kafka:                  kp,
+				cluster:                c.cfg.ID,
+				topic:                  topic,
+				diagnostic:             diagnostic,
 			}
 			w.Open()
```

Comment:
Can `w.Open` fail?  Should we check for an error?

Reply:
It can't fail, and there is no error returned